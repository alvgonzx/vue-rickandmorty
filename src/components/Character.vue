// character component
<script setup></script>

<template>
    <article class="w-full p-4">
        <div class="bg-white rounded-lg shadow-lg">
            <div class="bg-gray-400 h-64 rounded-t-lg"></div>
            <div class="p-4">
                <h2 class="text-2xl font-bold">Rick Sanchez</h2>
                <p class="text-gray-600">Human Â· Alive</p>
            </div>
        </div>
    </article>
</template>
