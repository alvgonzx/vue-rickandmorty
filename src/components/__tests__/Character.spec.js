import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";

import Character from "@/components/Character.vue";

describe("Character", () => {
    it("renders correctly", () => {
        const wrapper = mount(Character, {
            props: { name: "Rick Sanchez", status: "Alive", species: "Human", img: "https://rickandmortyapi.com/api/character/avatar/1.jpeg" },
        });
        expect(wrapper.html()).toMatchSnapshot();
    });
});
