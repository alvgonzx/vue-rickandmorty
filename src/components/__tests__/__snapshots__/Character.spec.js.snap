// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Character > renders correctly 1`] = `
"<article class=\\"w-full xl:w-1/4 p-4\\">
  <div class=\\"bg-white rounded-lg shadow-lg\\">
    <div class=\\"h-64 rounded-t-lg flex justify-center\\"><img src=\\"https://rickandmortyapi.com/api/character/avatar/1.jpeg\\"></div>
    <div class=\\"p-4 xl:text-left text-center\\">
      <h2 class=\\"text-2xl font-bold\\">Rick Sanchez</h2>
      <p class=\\"text-gray-600\\">Human Â· Alive</p>
    </div>
  </div>
</article>"
`;
